docs/blocks/blocks.md

```diff
@@ -35,6 +35,39 @@ Blocks must be [registered][registering] to function.
 
     When a block is registered, *only* a block is registered. The block does not automatically have an `ItemBlock`. To create a basic `ItemBlock` for a block one should use `new ItemBlock(block).setRegistryName(block.getRegistryName())`. The unlocalized name is the same as the block's. Custom subclasses of `ItemBlock` may be used as well. Once an `ItemBlock` has been registered for a block, `Item.getItemFromBlock` can be used to retrieve it. `Item.getItemFromBlock` will return `null` if there is no ItemBlock for the Block, so if you are not certain that there is an ItemBlock for the Block you are using, check for null.
 
+Coloring a Block
+----------------
+
+There can be blocks, textures of which are colored within the code, and in fact, there are a lot of such blocks. For instance, color of grass, vines, waterlilys and etc. changes depending on the biome it's placed in.
+
+### Block Color Handlers
+
+A block color handler is required to color a block and is an instance of `IBlockColor`, and therefore implements the following method.
+
+```java
+int colorMultiplier(
+  IBlockState state, 
+  @Nullable IBlockAccess worldIn, 
+  @Nullable BlockPos pos, 
+  int tintIndex)
+```
+
+This method returns a hex representation of a color in an integer. 
+
+Objects passed to the method make it possible to change color multiplier dynamically. `BlockModelRenderer#renderModelBrightness(IBakedModel, IBlockState, float, boolean)` passes null `worldIn` and `pos` calling the method. 
+
+Tint indexes are specified for faces of an element in block's model JSON file. When a face with a tint index is being drawn, its tint index is passed to the method. A face without a tint index won't be colored. Block's digging and hitting particles are hardcoded to have zeroth tint index.
```

Comment:
"When a face with a tint index is being drawn, its tint index is passed to the method. A face without a tint index won't be colored."

The first sentence is not necessary here. This could just be "A face without a tint index won't be colored, and therefore will not have its color handler called." or something